docs/proposals/improve-status-in-CAPI-resources.md

```diff
@@ -0,0 +1,958 @@
+| `BootstrapReady`               | `Initialization.BootstrapSecretCreated` (renamed)        | `Initialization.BootstrapSecretCreated`    |
+| `InfrastructureReady`          | `Initialization.InfrastructureProvisioned` (renamed)     | `Initialization.InfrastructureProvisioned` |
+|                                | `BackCompatibilty` (new)                                 | (removed)                                  |
+| `Phase` (deprecated)           | `BackCompatibilty.Phase` (renamed) (deprecated)          | (removed)                                  |
+| `LastUpdated` (deprecated)     | `BackCompatibilty.LastUpdated` (renamed) (deprecated)    | (removed)                                  |
+| `FailureReason` (deprecated)   | `BackCompatibilty.FailureReason` (renamed) (deprecated)  | (removed)                                  |
+| `FailureMessage` (deprecated)  | `BackCompatibilty.FailureMessage` (renamed) (deprecated) | (removed)                                  |
+| `Conditions`                   | `BackCompatibilty.Conditions` (renamed) (deprecated)     | (removed)                                  |
+| `ExperimentalConditions` (new) | `Conditions` (renamed)                                   | `Conditions`                               |
+| other fields...                | other fields...                                          | other fields...                            |
+
+Notes:
+- The `BackCompatibilty` struct is going to exist in v1beta2 types only until v1beta1 removal (9 months or 3 minor releases after v1beta2 is released/v1beta1 is deprecated, whichever is longer).
+  Fields in this struct are used for supporting down conversions, thus providing users relying on v1beta1 APIs additional buffer time to pick up the new changes. 
+
+##### Machine (New)Conditions
+
+| Condition              | Note                                                                                                                                                                                                                                                            |
+|------------------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `Available`            | True if at the machine is Ready for at least MinReady seconds, as defined by the Machine's minReadySeconds field                                                                                                                                                |
+| `Ready`                | True if Machine's `BootstrapSecretReady`, `InfrastructureReady`, `NodeHealthy` and `HealthCheckSucceeded` (if present) are true; if other conditions are defined in `spec.readinessGates`, those conditions should be true as well for the Machine to be ready. |
```

Comment:
@sbueringer, folks, opinions on making this condition to turn false as soon as the deletion process start?

Reply:
I think it should stay true. Just because a deletionTimestamp is set it doesn't mean that the Machine already gets deleted (deletion hooks, drain, ...). Machines could stay around for days after the deletionTimestamp being set without even reaching the stage where the Node is cordoned (aka they are still fully usable at this point)

Reply:
That's a fair point.
I will take a look if this might have an impact e.g. on how MD determine scale up or down.

@sbueringer follow up question, should we consider deleteting machines available or not (I think not, it is also probably in line on how MD works today, but I have to double check)

Reply:
I see deleting as orthogonal to both ready and available. As long as Machines are ready & available I think we should surface this accordingly (even if the deletion is ongoing). Once we reach the point where the Machine becomes un-ready / un-available because of deletion we would update accordingly.

I'm fine if we want e.g. to say: "as soon as we actually start draining a Machine becomes un-ready / un-available". Because at this point it can actually not be used anymore to schedule new Pods / workload

Reply:
With the definition of "machine can host workloads" it would be correct to mark a Machine as not ready as soon as we cordon. WDYT?

Reply:
Discussed this with Fabrizio & Christian. We now think it's okay to consider a Machine with deletionTimestamp is not ready. Even if the Machine deletion is "stuck" at the "pre-drain.delete" hook we have to assume that folks are already "cleaning up" / "terminating" the Pod in these hooks. We'll update the Ready condition accordingly